internal/contour/listener.go

```diff
@@ -403,6 +403,33 @@ func (v *listenerVisitor) visit(vertex dag.Vertex) {
 		v.listeners[ENVOY_HTTPS_LISTENER].FilterChains = append(v.listeners[ENVOY_HTTPS_LISTENER].FilterChains,
 			envoy.FilterChainTLS(vh.VirtualHost.Name, downstreamTLS, filters))
 
+		// If this VirtualHost has enabled the fallback certificate then set a default FilterChain which will allow
+		// routes with this vhost to accept non SNI TLS requests
+		if vh.FallbackCertificate != nil && !envoy.ContainsFallbackFilterChain(v.listeners[ENVOY_HTTPS_LISTENER].FilterChains) {
+
+			// Choose the higher of the configured or requested TLS version.
+			vers := max(v.ListenerVisitorConfig.minProtoVersion(), vh.MinProtoVersion)
+
+			downstreamTLS = envoy.DownstreamTLSContext(
+				vh.FallbackCertificate,
```

Comment:
this needs to be added to secret cache here https://github.com/projectcontour/contour/blob/ea65fa53a7d3feae99713bfb253a04b1671630b8/internal/contour/secret.go#L88
for it to be available through SDS cache ?

Reply:
the secret should already be in the cache since it's in Kubernetes.